File: app/src/main/java/lt/vilnius/tvarkau/dagger/module/AnalyticsModule.java
package lt.vilnius.tvarkau.dagger.module;

import android.app.Application;

import com.mixpanel.android.mpmetrics.MixpanelAPI;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import lt.vilnius.tvarkau.R;
import lt.vilnius.tvarkau.analytics.Analytics;
import lt.vilnius.tvarkau.analytics.RemoteAnalytics;

@Module
public class AnalyticsModule {

    @Provides
    @Singleton
    MixpanelAPI providesMixpanel(Application application) {
        return MixpanelAPI.getInstance(application, application.getString(R.string.mixpanel_token));
    }

    @Provides
    @Singleton
    Analytics providesAnalytics(Application application, MixpanelAPI mixpanelAPI) {
        return new RemoteAnalytics(application.getApplicationContext(), mixpanelAPI);
    }
}

File: app/src/main/java/lt/vilnius/tvarkau/dagger/module/AnalyticsModule.kt
package lt.vilnius.tvarkau.dagger.module

import com.mixpanel.android.mpmetrics.MixpanelAPI
import dagger.Module
import dagger.Provides
import lt.vilnius.tvarkau.R
import lt.vilnius.tvarkau.TvarkauApplication
import lt.vilnius.tvarkau.analytics.Analytics
import lt.vilnius.tvarkau.analytics.RemoteAnalytics
import javax.inject.Singleton

@Module
class AnalyticsModule {

    @Provides
    @Singleton
    fun providesMixpanel(application: TvarkauApplication): MixpanelAPI {
        return MixpanelAPI.getInstance(application, application.getString(R.string.mixpanel_token))
    }

    @Provides
    @Singleton
    fun providesAnalytics(application: TvarkauApplication, mixpanelAPI: MixpanelAPI): Analytics {
        return RemoteAnalytics(application.applicationContext, mixpanelAPI)
    }
}
